packages/headless/src/features/generated-answer/generated-answer-analytics-actions.ts

```diff
@@ -0,0 +1,60 @@
+import {GeneratedAnswerCitation} from '../../api/generated-answer/generated-answer-event-payload';
+import {
+  AnalyticsType,
+  CustomAction,
+  SearchAction,
+  makeAnalyticsAction,
+} from '../analytics/analytics-utils';
+
+export const logRetryGeneratedAnswer = (): SearchAction =>
+  makeAnalyticsAction(
+    'generatedAnswer/logRetryGeneratedAnswer',
+    AnalyticsType.Search,
+    (client) => client.makeRetryGeneratedAnswer()
+  );
+
+export const logOpenGeneratedAnswerSource = (
+  citation: GeneratedAnswerCitation
+): CustomAction =>
+  makeAnalyticsAction(
+    'generatedAnswer/logOpenGeneratedAnswerSource',
+    AnalyticsType.Custom,
+    (client, state) => {
+      const generativeQuestionAnsweringId =
+        state.search?.response?.extendedResults
+          ?.generativeQuestionAnsweringId ?? '';
+      return client.makeOpenGeneratedAnswerSource({
+        generativeQuestionAnsweringId,
+        permanentId: citation.permanentid,
+        id: citation.id,
```

Comment:
What's the difference between the two Ids?

Each citation will have an individual ID even if the permanentID of the document behind it is the same?

Reply:
Yeah the citation id appears to be something to identify the citation used by ML. Given it's a different instance than the document itself I think it makes sense to keep both.

Reply:
All good was just making sure it was clear